app/controllers/api/discussions_controller.rb

```diff
@@ -46,13 +47,15 @@ def set_volume
   private
 
   def visible_records
-    Queries::VisibleDiscussions.new(user: current_user, groups: visible_groups)
+    Queries::VisibleDiscussions.new(user: current_user, groups: organisation_groups || current_user.groups)
   end
 
-  def visible_groups
-    return current_user.groups unless params[:group_id] || params[:group_key]
-    load_and_authorize :group
-    [@group, @group.subgroups].flatten
+  def public_records
+    Queries::VisibleDiscussions.new(user: current_user, groups: organisation_groups || Group.visible_to_public)
+  end
+
+  def organisation_groups
+    [@group, @group.subgroups].flatten if @group
```

Comment:
Here's another one where we're going 'if there's a group, give me just the ones in that group, otherwise give me everything for this user / everything public'